### PR TITLE
Antd v4

### PR DESCRIPTION
**Pull request by Monteth**
This PR includes an update of the AntDesign uniforms theme to version `4.X.X`. This change is braking and drops support for older major versions (`2.X`, `3.X`).

### CODE REVIEW

**Comment by Monteth** [comment]
@pro-kswe Thanks for the improvement suggestion and kind words!

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "test": "jest"
   },
   "dependencies": {
+    "@ant-design/icons": "^4.2.1",
```
Comment (radekmie) [inline package.json:20]: Pin this version.
```suggestion
    "@ant-design/icons": "4.2.1",
```

File: packages/uniforms-antd/__tests__/LongTextField.tsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
-import Input from 'antd/lib/input';
 import React from 'react';
 import { LongTextField } from 'uniforms-antd';
+import TextArea from 'antd/lib/input/TextArea';
```
Comment (radekmie) [inline packages/uniforms-antd/__tests__/LongTextField.tsx:3]: Keep imports ordered.
```suggestion
import React from 'react';
import TextArea from 'antd/lib/input/TextArea';
import { LongTextField } from 'uniforms-antd';
```

File: packages/uniforms-antd/src/BaseForm.tsx [hunk 3]
```diff
@@ -1,10 +1,20 @@
 import { BaseForm } from 'uniforms';
+import classnames from 'classnames';
```
Comment (radekmie) [inline packages/uniforms-antd/src/BaseForm.tsx:2]: Keep imports ordered. I won't add a comment for each re-ordering, just not to spam the comments.

File: packages/uniforms-antd/package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
     "src/*.tsx"
   ],
   "peerDependencies": {
-    "antd": "^3.0.0 || ^2.7.0",
+    "antd": "^4.0.0",
```
Comment (radekmie) [inline packages/uniforms-antd/package.json:29]: We need to include `@ant-design/icons` here as well.

File: packages/uniforms-antd/src/BoolField.tsx [hunk 5]
```diff
@@ -12,6 +13,7 @@ export type BoolFieldProps = Override<
     checkbox?: boolean;
     inputRef?: Ref<HTMLInputElement>;
     onChange(value?: boolean): void;
+    onReset?: () => void;
```
Comment (radekmie) [inline packages/uniforms-antd/src/BoolField.tsx:16]: Why is this change needed?

File: website/package.json [hunk 6]
```diff
@@ -10,7 +10,7 @@
     "@docusaurus/preset-classic": "2.0.0-alpha.58",
     "@material-ui/core": "4.9.9",
     "ajv": "6.12.0",
-    "antd": "3.26.15",
+    "antd": "4.4.0",
```
Comment (radekmie) [inline website/package.json:13]: Icons package is missing.

File: scripts/setupEnzyme.ts [hunk 7]
```diff
@@ -1,4 +1,18 @@
...
+  writable: true,
+  value: jest.fn().mockImplementation(query => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // deprecated
+    removeListener: jest.fn(), // deprecated
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
```
Comment (radekmie) [inline scripts/setupEnzyme.ts:14]: 1. Let's sort them.
2. We don't use that for the tests, therefore lets put no-op functions here. `jest.fn` record all calls and will increase memory usage.

File: packages/uniforms-antd/src/DateField.tsx [hunk 8]
```diff
@@ -2,7 +2,7 @@ import DatePicker from 'antd/lib/date-picker';
 import React, { Ref } from 'react';
 import moment from 'moment';
 import { connectField, filterDOMProps, Override } from 'uniforms';
-import { DatePickerProps } from 'antd/lib/date-picker/interface';
+import { DatePickerProps } from 'antd/es/date-picker';
```
Comment (radekmie) [inline packages/uniforms-antd/src/DateField.tsx:5]: Are types exported from the `'antd'` root package? We could use that, as TypeScript removes type-only imports anyway.

File: packages/uniforms-antd/src/BoolField.tsx (hunk 5 above)
Comment (Monteth) [reply packages/uniforms-antd/src/BoolField.tsx:16]: Removing this line cause 
```ts
Type '(event: FormEvent<HTMLDivElement>) => void' is not assignable to type '() => void'
```
in wrapper props.
Seems like `FormItemProps` does not satisfy all input fields.

Comment (radekmie) [reply packages/uniforms-antd/src/BoolField.tsx:16]: Ah, OK. Use `Omit<HTMLProps<HTMLDivElement>, 'onReset'>` then.

File: packages/uniforms-antd/src/DateField.tsx (hunk 8 above)
Comment (Monteth) [reply packages/uniforms-antd/src/DateField.tsx:5]: No, they are not available there. We can import `DatePickerProps` from `antd/lib/date-picker` just like `DatePicker` what I missed.

File: packages/uniforms-antd/package.json (hunk 4 above)
Comment (radekmie) [reply packages/uniforms-antd/package.json:29]: And keep them ordered :stuck_out_tongue:

File: packages/uniforms-antd/src/DateField.tsx [hunk 9]
```diff
@@ -1,8 +1,7 @@
-import DatePicker from 'antd/lib/date-picker';
-import React, { Ref } from 'react';
+import DatePicker, { DatePickerProps } from 'antd/lib/date-picker';
 import moment from 'moment';
+import React, { Ref } from 'react';
```
Comment (radekmie) [inline packages/uniforms-antd/src/DateField.tsx:3]:
```suggestion
import DatePicker, { DatePickerProps } from 'antd/lib/date-picker';
import React, { Ref } from 'react';
import moment from 'moment';
```

File: packages/uniforms-antd/src/ListAddField.tsx [hunk 10]
```diff
@@ -1,12 +1,13 @@
 import Button, { ButtonProps, ButtonSize, ButtonType } from 'antd/lib/button';
-import React from 'react';
 import cloneDeep from 'lodash/cloneDeep';
+import PlusSquareOutlined from '@ant-design/icons/PlusSquareOutlined';
+import React from 'react';
```
Comment (radekmie) [inline packages/uniforms-antd/src/ListAddField.tsx:4]:
```suggestion
import PlusSquareOutlined from '@ant-design/icons/PlusSquareOutlined';
import React from 'react';
import cloneDeep from 'lodash/cloneDeep';
```

File: packages/uniforms-antd/src/LongTextField.tsx [hunk 11]
```diff
@@ -1,19 +1,14 @@
-import Input, { InputProps, TextAreaProps } from 'antd/lib/input';
 import React, { Ref } from 'react';
+import TextArea from 'antd/lib/input/TextArea';
 import { connectField, filterDOMProps, Override } from 'uniforms';
+import { TextAreaProps } from 'antd/lib/input';
```
Comment (radekmie) [inline packages/uniforms-antd/src/LongTextField.tsx:4]:
```suggestion
import { TextAreaProps } from 'antd/lib/input';
import { connectField, filterDOMProps, Override } from 'uniforms';
```

**Comment by Monteth** [comment]
OK, now I changed my local script to case sensitive sort. All I have to do from now is remember to run it every time...
